src/gui/socketapi.cpp

```diff
@@ -366,8 +396,20 @@ void SocketApi::broadcastMessage(const QString &msg, bool doWait)
     }
 }
 
-void SocketApi::processShareRequest(const QString &localFile, SocketListener *listener, ShareDialogStartPage startPage)
+void SocketApi::processShareRequest(const QString &localFileC, SocketListener *listener, ShareDialogStartPage startPage)
 {
+    QString localFile = localFileC;
+
+#if defined(Q_OS_WIN)
+    OCC::ConfigFile Cfg;
+    char letter[2];
+    letter[0] = localFile.toStdString().c_str()[0];
+    letter[1] = 0;
+
+    if (!QString(letter).compare(Cfg.defaultFileStreamLetterDrive().toUpper()))
```

Comment:
If we assume it's always a single letter (as the code above implies), defaultFileStreamLetterDrive could return a QChar then I guess.

But is it always a single letter?

Also that whole logic seems to be present plenty of times in that file, probably needs to be factored out.